graphula-core/src/Graphula/Arbitrary.hs

```diff
@@ -0,0 +1,34 @@
+{-|
+  Graphula tracks its own 'QCGen' for deterministic generation with 'Arbitrary'
+  and 'Gen'. 'generate' can be used to produce arbitrary values utilizing
+  graphula's generation.
+-}
+module Graphula.Arbitrary
+  ( generate
+  )
+where
+
+import Prelude
+
+import Control.Monad.IO.Unlift (MonadIO, liftIO)
+import Data.IORef (readIORef, writeIORef)
+import Graphula.Internal (MonadGraphulaBackend, askGen)
+import System.Random (split)
+import Test.QuickCheck (Gen)
+import Test.QuickCheck.Gen (unGen)
+
+-- | Run a generator
+--
+-- This is akin to 'Test.QuickCheck.generate', but utilizing graphula's
+-- generation. The size passed to the generator is always 30; if you want
+-- another size then you should explicitly use 'Test.QuickCheck.resize'.
+--
+generate :: (MonadIO m, MonadGraphulaBackend m) => Gen a -> m a
```

Comment:
Neat. We'll need to remember to replace the ad-hoc `liftIO $ generate arbitrary`'s we have in our tests for, e.g. `UUID`s and emails.